mllib/src/main/scala/org/apache/spark/mllib/feature/PCA.scala

```diff
@@ -41,6 +41,7 @@ class PCA @Since("1.4.0") (@Since("1.4.0") val k: Int) {
    */
   @Since("1.4.0")
   def fit(sources: RDD[Vector]): PCAModel = {
+    sources.persist()
```

Comment:
`sources` is not some intermediate results, it is just user input. Hence removing persist from here.
https://github.com/apache/spark/blob/80fbc382a60973db21367a922a0fb797e5ab382d/mllib/src/main/scala/org/apache/spark/ml/feature/PCA.scala#L95